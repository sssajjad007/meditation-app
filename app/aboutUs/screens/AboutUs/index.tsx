import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React from "react";
import { View, Image, Linking } from "react-native";
import {
  Button,
  Container,
  THEME,
  Typography,
  version,
} from "../../../library";
import * as Analytics from "expo-firebase-analytics";
import { styles } from "./styles";
import Communications from "react-native-communications";
import { Scroller } from "../../../library/Scroller";
import { digitsEnToFa } from "@persian-tools/persian-tools";
export function AboutUs() {
  async function callEvent() {
    try {
      await Analytics.logEvent("  customer_service_call");
    } catch (error) {
      // console.log(error);
    }
  }
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container headerTitle={"درباره ما"}>
      <Scroller>
        <View style={styles.iconContainer}>
          <Image
            source={require("../../assets/Logo.png")}
            style={styles.imageSize}
            resizeMode={"contain"}
          />
        </View>
        <View style={styles.bodyContainer}>
          <Typography
            mode={"mediumBody18"}
            style={{ color: THEME.COLORS.LABEL.SECONDARY, textAlign: "right" }}
          >
            {`دنج اپلیکیشنی برای سلامتی ذهن و بدن شماست. در شرایط زندگی امروزی پیدا کردن زمانی برای اینکه قدری تمرکز کنیم و کمتر در افکار غرق باشیم تبدیل شده به یه کار خیلی سخت. ما باور داریم خالی کردن ۱۰ دقیقه در روز برای مدیتیشن امکان‌پذیره. هدف ما در دنج اینه که این فرصت رو به بهترین شکل براتون ایجاد کنیم. در دنج از اپ‌های Calm، Waking Up و Headspace استفاده شده و ترجمه تمام مدیتیشن‌ها در تیم دنج انجام گرفته. برخی از جلسات هم توسط تیم دنج نوشته شده‌اند. با دنج لذت حضور رو تجربه کنید. `}
          </Typography>
        </View>
        <View style={styles.buttonContianer}>
          <Button
            mode={"text"}
            rippleColor={"lightGrey"}
            size={"small"}
            onPress={() => {
              navigation.push("TermsScreen");
            }}
          >
            <Typography
              mode={"boldBody16"}
              style={{ color: THEME.COLORS.PRIMARY.NORMAL }}
            >
              {"قوانین دنج"}
            </Typography>
          </Button>
          <Typography
            mode={"regularSub12"}
            style={{ color: THEME.COLORS.LABEL.TERTIARY, bottom: 12 }}
          >
            {`نسخه ${digitsEnToFa(version)}`}
          </Typography>
          <Button
            mode={"contained"}
            size={"big"}
            rippleColor={"lightGrey"}
            onPress={async () => {
              await callEvent();
              Linking.openURL("tel:02191006901");
              // Communications.phonecall("02191006901", false);
            }}
          >
            {"تماس با ما"}
          </Button>
        </View>
      </Scroller>
    </Container>
  );
}
