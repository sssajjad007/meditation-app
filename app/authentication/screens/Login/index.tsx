import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React from "react";
import { View, Image } from "react-native";
import { Button, Typography, THEME, Container } from "../../../library";
import { LogoImage } from "../../assets";
import { styles } from "./styles";

export function Login() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container style={styles.containerStyles}>
      <View style={{ flex: 1 }}>
        <Image style={styles.imageSize} source={LogoImage} />
      </View>
      <View style={{ flex: 0.5, justifyContent: "center" }}>
        <Typography
          mode={"mediumHeadline22"}
          style={{
            color: THEME.COLORS.LABEL.PRIMARY,
          }}
        >
          {"به دنج خوش اومدید !"}
        </Typography>
      </View>
      <View style={{ flex: 1 }}>
        <Typography
          mode={"mediumBody16"}
          style={{ color: THEME.COLORS.LABEL.SECONDARY }}
        >
          {"با دنج میتونید مدیتیشن رو یاد بگیرید و تمرین کنید."}
        </Typography>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          mode={"contained"}
          rippleColor={"lightGrey"}
          size={"big"}
          onPress={() => {
            navigation.push("Auth");
          }}
        >
          {"ورود"}
        </Button>
      </View>
    </Container>
  );
}
