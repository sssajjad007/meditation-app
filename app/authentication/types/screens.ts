export type tStep = "phone" | "genCode" | "name" | "genCodeResend"|"enterGenCode";
