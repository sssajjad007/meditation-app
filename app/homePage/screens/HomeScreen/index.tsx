import React, {
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from "react";
import * as Analytics from "expo-firebase-analytics";
import {
  View,
  Image,
  ActivityIndicator,
  BackHandler,
  Modal,
} from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { useBackHandler } from "@react-native-community/hooks";
import { Button, CustomBackdrop, storage, Typography } from "../../../library";
import {
  BottomSheetModal,
  BottomSheetView,
  BottomSheetBackdrop,
} from "@gorhom/bottom-sheet";
import { styles } from "./styles";
import { useIsFocused, useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { heightPercentageToDP } from "react-native-responsive-screen";
import { getAllEmergencyMeditation, getAllTracks } from "../../usecases";
import { ProfileIcon, ShopBoldIcon, HomeShopIcon } from "../../../library/Icon";
import { digitsEnToFa } from "@persian-tools/persian-tools";
import { getIsPremium } from "../../usecases/getIsPremium";
import { SafeAreaView } from "react-native-safe-area-context";
import {
  DailyMeditation,
  DiscountCardComponent,
  InstagramCardComponent,
  IntroductionMeditation,
  ExitSheet,
} from "../../components";
import { IosAddToHome } from "../../components/IosAddToHome";
import { UAParser } from "ua-parser-js";
export function HomeScreen() {
  const [modalVisible, setModalVisible] = useState<boolean>(false);
  const isFocused = useIsFocused();
  const isSet = useRef(false);
  const [loading, setLoading] = useState<boolean>(true);
  const [emergencyCourse, setEmergencyCourse] = useState([]);
  const [state, setState] = useState({
    course: [""],
    courseId: "",
    description: "",
    poster: "",
    title: "",
  });
  const snapPointsModal = useMemo(() => ["100%", "100%"], []);
  const BottomSheetModalRef = useRef<BottomSheetModal>(null);
  const closeModal = useCallback(() => {
    // for ios modal
    BottomSheetModalRef.current?.close();
  }, []);
  const onCollapsePressModal = useCallback(() => {
    //for ios modal
    BottomSheetModalRef.current?.present();
  }, []);

  useBackHandler(() => {
    if (!isFocused) return false;
    if (exitModalOpen.current) {
      closeExitModalPress();
      return true;
    }
    handleExitModalPress();
    return true;
  });

  async function init() {
    await Promise.all([getAllTracks(), EmergencyMeditation()]);
    setLoading(false);
  }
  async function shopEvent() {
    try {
      await Analytics.logEvent("store_popup_click");
    } catch (error) {
      console.log(error);
    }
  }
  async function shopSkipEvent() {
    try {
      await Analytics.logEvent("store_popup_skipped");
    } catch (error) {
      console.log(error);
    }
  }
  async function profileEvent() {
    try {
      await Analytics.logEvent("profile_home_click");
    } catch (error) {
      console.log(error);
    }
  }
  async function homeTabEvent() {
    try {
      await Analytics.logEvent("navigationbar_home_click");
      // console.log("homeTabEvent");
    } catch (error) {
      console.log(error);
    }
  }

  async function EmergencyMeditation() {
    const { emergencyMeditation } = await getAllEmergencyMeditation();
    if (emergencyMeditation) {
      setState({
        course: emergencyMeditation.course,
        courseId: emergencyMeditation.objectId,
        description: emergencyMeditation.description,
        poster: emergencyMeditation.poster,
        title: emergencyMeditation.title,
      });
      const result: any = [];
      for (let index = 0; index < emergencyMeditation.course.length; index++) {
        if (emergencyMeditation.course[index]) {
          const allCourse = storage.retrieve(
            emergencyMeditation.course[index],
            "string"
          );
          if (allCourse) {
            result.push(JSON.parse(allCourse));
          }
        }
      }
      setEmergencyCourse(result);
    }
  }

  function shopPopUpShown() {
    const showIosModal = storage.retrieve("show_ios_modal", "boolean");
    const isPremium = storage.retrieve("is_premium", "boolean");
    const shopPopUpShown = storage.retrieve("shop_pop_up_shown", "boolean");
    if (shopPopUpShown === true && isPremium === false) {
      storage.add("shop_pop_up_shown", false);
      handlePresentModalPress(); //shop
    }

    if (!showIosModal) {
      storage.add("show_ios_modal", true);
      // use ua parser safely
      const ua = new UAParser();
      ua.setUA(navigator?.userAgent);
      if (ua.getOS().name === "iOS") {
        setModalVisible(true);

        // onCollapsePressModal(); //ios
      }
    }
    return;
  }
  async function getData() {
    await getIsPremium();
    shopPopUpShown();
  }
  useEffect(() => {
    if (isFocused) {
      homeTabEvent();
      getData();
      if (!isSet.current) {
        init();
        isSet.current = true;
      }
    }
  }, [isFocused]);

  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const bottomSheetModalRef = useRef<BottomSheetModal>(null);
  const exitSheetModalRef = useRef<BottomSheetModal>(null);
  const exitModalOpen = useRef<boolean>(false);

  function handleExitModalPress() {
    //for exit modal
    exitModalOpen.current = true;
    exitSheetModalRef.current?.present();
  }
  function closeExitModalPress() {
    //for exit modal
    exitModalOpen.current = false;
    exitSheetModalRef.current?.close();
  }
  const snapPoints = useMemo(() => [270, 270], []);
  const handlePresentModalPress = useCallback(() => {
    //for shop modal
    bottomSheetModalRef.current?.present();
  }, []);
  const handleSnapPress = useCallback(() => {
    //for shop modal
    bottomSheetModalRef.current?.close();
  }, []);
  function onPress() {
    setModalVisible(!modalVisible);
  }
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
      >
        <View style={styles.header}>
          <View style={styles.iconsContainer}>
            <HomeShopIcon
              size={24}
              onPress={() => {
                navigation.push("ShopScreen");
              }}
            />
            <ProfileIcon
              size={24}
              onPress={async () => {
                await profileEvent();
                navigation.push("Profile");
              }}
            />
          </View>

          <Typography mode={"regularBody16"} style={{ textAlign: "right" }}>
            {"مدیتیشن روزانه"}
          </Typography>
        </View>
        <DailyMeditation loading={loading} />
        <DiscountCardComponent />
        <IntroductionMeditation loading={loading} />
        <View style={styles.emergency}>
          <View style={styles.emergencyBody}>
            <Typography mode={"regularSubhead20"}>
              {"مدیتیشن اضطراری"}
            </Typography>
            <View style={styles.emergencyDetails}>
              <Typography mode={"lightSub10"}>
                {emergencyCourse.length === 0
                  ? `-- جلسه`
                  : `${digitsEnToFa(emergencyCourse.length)} جلسه`}
              </Typography>
              <View style={styles.lineVertical} />
              <Typography mode={"boldSub12"}>{"خستگی، خشم، ترس"}</Typography>
            </View>
            <View style={styles.emergencyDetails}>
              <Button
                mode={"text"}
                rippleColor={"lightGrey"}
                size={"small"}
                onPress={() => {
                  navigation.push("MeditationCourse", {
                    objectId: state.courseId,
                    track: state.course,
                    title: state.title,
                    poster: state.poster,
                    description: state.description,
                  });
                }}
              >
                {"مشاهده همه"}
              </Button>
            </View>
          </View>
          <View style={styles.emergencyCard}>
            {state.poster ? (
              <Image
                // source={{ uri: introductionPoster }}
                source={{ uri: state.poster }}
                style={styles.emergencyImage}
                resizeMode={"contain"}
              />
            ) : (
              <ActivityIndicator />
            )}
          </View>
        </View>
        <InstagramCardComponent />
        <BottomSheetModal
          ref={bottomSheetModalRef}
          index={1}
          backdropComponent={(props) => (
            <BottomSheetBackdrop pressBehavior={"close"} {...props} />
          )}
          snapPoints={snapPoints}
          enablePanDownToClose={false}
          enableHandlePanningGesture={false}
          handleIndicatorStyle={{ backgroundColor: "transparent" }}
        >
          <BottomSheetView
            style={{
              flex: 1,
              alignItems: "center",
            }}
          >
            <View style={styles.bottomSheetHeader}>
              <View style={styles.bottomSheetTitle}>
                <ShopBoldIcon size={32} />
                <Typography mode={"mediumBody18"} style={{ paddingRight: 8 }}>
                  {"دنج پلاس"}
                </Typography>
              </View>
              <View style={styles.lineHorizontal} />
            </View>
            <View
              style={{
                width: "90%",
                flexDirection: "column",
              }}
            >
              <Typography mode={"mediumBody18"}>
                {"برای دسترسی به تمام دوره ها اشتراک دنج را خریداری کنید."}
              </Typography>
              <View
                style={{
                  flexDirection: "row-reverse",
                  alignItems: "center",
                  marginTop: heightPercentageToDP(2),
                  justifyContent: "space-evenly",
                }}
              >
                <Button
                  mode={"contained"}
                  size={"xMedium"}
                  rippleColor={"lightGrey"}
                  onPress={async () => {
                    await shopEvent();
                    navigation.push("ShopScreen");
                    handleSnapPress();
                  }}
                >
                  {"برو به فروشگاه"}
                </Button>
                <Button
                  mode={"outlined"}
                  size={"xSmall"}
                  rippleColor={"lightGrey"}
                  onPress={async () => {
                    await shopSkipEvent();
                    handleSnapPress();
                  }}
                >
                  {"فعلا نه"}
                </Button>
              </View>
            </View>
          </BottomSheetView>
        </BottomSheetModal>
        <BottomSheetModal
          ref={exitSheetModalRef}
          index={1}
          backdropComponent={(props) => (
            <BottomSheetBackdrop pressBehavior={"close"} {...props} />
          )}
          snapPoints={snapPoints}
          enablePanDownToClose={false}
          enableHandlePanningGesture={false}
          handleIndicatorStyle={{ backgroundColor: "transparent" }}
        >
          <ExitSheet
            onCancelPress={closeExitModalPress}
            onExitPress={() => {
              BackHandler.exitApp();
            }}
          />
        </BottomSheetModal>
        <Modal
          // ref={BottomSheetModalRef}
          // snapPoints={snapPointsModal}
          // backdropComponent={CustomBackdrop}
          // detached={true}
          // bottomInset={0}
          // style={styles.modal}
          // index={1}
          // enablePanDownToClose={false}
          // handleIndicatorStyle={{ backgroundColor: "transparent" }}
          animationType="slide"
          transparent={false}
          visible={modalVisible}
          // onRequestClose={() => {
          //   setModalVisible(!modalVisible);
          // }}
        >
          <IosAddToHome CloseModal={onPress} />
        </Modal>
      </ScrollView>
    </SafeAreaView>
  );
}
