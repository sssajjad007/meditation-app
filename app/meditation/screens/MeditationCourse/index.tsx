import { useNavigation, useRoute } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { useEffect, useState } from "react";
import { View, Image } from "react-native";
import { Container, storage, THEME, Typography } from "../../../library";
import { Scroller } from "../../../library/Scroller";
import { MeditationCardComponent } from "../../components/MeditationCard";
import { styles } from "./styles";

export function MeditationCourse() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const [course, setCourse] = useState([]);
  const routes = useRoute<any>();
  const { track, title, poster, description, objectId } = routes.params;
  const [isPremium, setIsPremium] = useState<boolean>(false);
  async function retriveCourseDetails() {
    const isUSerPremium = storage.retrieve("is_premium", "boolean");
    if (isUSerPremium != undefined) {
      setIsPremium(isUSerPremium);
    }
    const result: any = [];
    for (let index = 0; index < track.length; index++) {
      if (track[index]) {
        const allCourse = storage.retrieve(track[index], "string");
        if (allCourse) {
          result.push(JSON.parse(allCourse));
        } else return;
      } else return;
    }
    setCourse(result);
  }
  function typeRender(title: string) {
    if (title.includes("پیاده")) {
      return "walk";
    }
    if (title.includes("اضطراب")) {
      return "stress";
    }
    if (title.includes("اضطراری")) {
      return "emergency";
    }
    if (title.includes("عزت")) {
      return "selfesteem";
    }
    if (title.includes("خواب")) {
      return "sleep";
    }
    if (title.includes("اسکن")) {
      return "bodyscan";
    }
    return "";
  }
  function renderer() {
    const result: JSX.Element[] = [];
    if (course.length === 0 || isPremium === undefined) {
      return (
        <Typography mode={"boldBody16"} style={{ textAlign: "right" }}>
          {"در حال بارگذاری..."}
        </Typography>
      );
    }
    for (let index = 0; index < course.length; index++) {
      let coursePremium = false;
      const element = course[index];
      if (isPremium === false) {
        if (course.length < 10) {
          if (index > Math.floor(course.length / 10)) {
            coursePremium = true;
          }
        } else {
          if (index >= Math.floor(course.length / 10)) {
            coursePremium = true;
          }
        }
      }
      result.push(
        <MeditationCardComponent
          title={element.title}
          key={element.objectId}
          duration={element.duration}
          coursePremium={coursePremium}
          onPress={() => {
            navigation.push("Player", {
              courseId: objectId,
              id: element.objectId,
              title: element.title,
              url: element.url,
              mode: "course",
              type: typeRender(title),
            });
          }}
        />
      );
    }
    return result;
  }
  useEffect(() => {
    retriveCourseDetails();
  }, []);

  return (
    <Container headerTitle={title}>
      <Scroller>
        <View style={{ minHeight: 700 }}>
          <View style={styles.assets}>
            <Image
              source={{ uri: poster }}
              style={{ width: "100%", height: "100%" }}
            />
          </View>
          <View style={styles.description}>
            <Typography
              mode={"regularSub12"}
              style={{
                textAlign: "right",
                color: THEME.COLORS.LABEL.SECONDARY,
              }}
            >
              {description}
            </Typography>
          </View>
          <View style={styles.cardContainer}>{renderer()}</View>
        </View>
      </Scroller>
    </Container>
  );
}
