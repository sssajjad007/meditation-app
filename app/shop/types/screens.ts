export interface IUseCafeBazaar {
  setSubscriptionDaysLeft: (daysLeft: number) => void;
  setUserPremium: (isPremium: boolean) => void;
}
