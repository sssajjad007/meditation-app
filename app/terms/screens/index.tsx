import { View, Text } from "react-native";
import React from "react";
import { styles } from "./styles";
import { Container, Paragraph, THEME, Typography } from "../../library";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { Scroller } from "../../library/Scroller";

export function TermsScreen() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container
      headerTitle={"قوانین دنج"}
      headerTitleColor={"white"}
      backgroundColor={THEME.COLORS.LABEL.PRIMARY}
    >
      <Scroller
        contentContainerStyle={{ backgroundColor: THEME.COLORS.LABEL.PRIMARY }}
      >
        {/* <View style={styles.header}>
        <Title style={styles.toRight}>{"قوانین دنج"}</Title>
        <View style={styles.backIcon}>
          <ArrowRightIcon
            size={30}
            onPress={() => {
              navigation.goBack();
            }}
          />
        </View>
      </View> */}
        <View style={styles.termContainere}>
          <Typography
            mode={"regularBody14"}
            style={{ color: "white", textAlign: "right" }}
          >
            {`ماده ۱. این سیاست‌نامه حفظ حریم خصوصی برای محصول دنج ارائه شده توسط شرکت کاوشگران رادین فناور سلامت با نام تجاری کرفس است. دنج یک اپلیکیشن مدیتیشن است که با هدف آموزش و تسهیل در تمرین مدیتیشن در بین فارسی‌زبانان ایجاد شده است.`}
            {`
          
ماده ۲. قبل از استفاده از برنامه کاربران ملزم به ثبت‌نام می‌باشند. ثبت نام در دنج رایگان است. در قسمت ثبت‌نام اسم و شماره موبایل کاربر پرسیده می‌شود. اطلاعات دیگر از جمله سن و جنسیت از قسمت پروفایل در اپلیکیشن دنج قابل تکمیل شدن است. جمع‌آوری این اطلاعات به دنج اجازه می‌دهد تا دید دقیق‌تری نسبت به کاربران خود داشته باشد تا علاوه بر تعامل راحت‌تر، امکانات مناسب‌تری را برای‌ آن‌ها فراهم کند. دنج این اطلاعات را بدون اجازه‌ی کاربر با هیچ شخص ثالثی در میان نخواهد گذاشت.`}
            {`

ماده ۳. کاربر، مسئول حفظ و نگهداری سخت‌افزار کامپیوتر، تلفن‌ها و سایر ابزار ضروری برای دسترسی و استفاده از این سایت و تمام هزینه‌های مرتبط به آن بوده و دنج مسئول صدمات ناشی از ابزار کاربر نیست.
`}
            {`
ماده ۴. دنج اقدامات امنیتی لازم و مناسبِ سخت‌افزاری و نرم‌افزاری را برای جلوگیری از دسترسی‌های غیرمجاز و افشاء یا تخریب اطلاعات شخصی کاربر انجام می‌دهد. دنج اجازه دسترسی به اطلاعات شخصی کاربران را به کارکنان این شرکت می‌دهد اما این افراد با تعهدات محرمانه مواجه هستند و در صورت تخطی از این تعهدات با پیگرد قانونی مواجه خواهند شد. همچنین مطالعات آماری در مورد جامعه‌ی کاربران دنج بدون استفاده از هویت‌ شخصی آن‌ها و کاملاً ناشناس انجام می‌شود.`}
            {`

ماده ۵. دنج ممکن است با شرکت‌ها، وبسایت‌ها یا اپلیکیشن‌های دیگر همکاری کند اما هیچ‌گونه مسئولیتی در برابر سیاست‌های حفظ حریم شخصی و کاربری آنها ندارد. در صورتی که تمایل دارید از طریق اپلیکیشن دنج به سایت‌ها یا اپلیکیشن‌های دیگر وارد شوید، برای اطلاع از سیاست‌های حفظ حریم خصوصی این مجموعه‌ها به وب‌سایت آنها مراجعه کنید.`}
            {`

ماده ۶. تنها افراد بالای ۱۸ سال حق انجام معامله (خرید) را در اپلیکیشن دنج دارند.`}
            {`

ماده ۷. محصول خریدنی در دنج اشتراک است که با تهیه آن کاربر به تمام محتوا و امکانات اپ دسترسی پیدا می‌کند.
`}

            {`
ماده ۸. فرایند خرید در دنج به شرح زیر است:
انتخاب بسته‌ی اشتراکی مورد نظر
مشاهده‌ی پیش‌فاکتور
پرداخت از طریق درگاه اینترنتی
دریافت بسته‌ی اشتراکی`}
            {`

ماده ۹. اختلال در سرویس اینترنت یا هر دلیلی که باعث شود فرایند خرید به انجام نرسد باعث از بین رفتن حق کاربر نمی‌شود و در صورتی که مبلغی از حساب کاربر کسر شده باشد از طریق بانک به وی بازپرداخته خواهد شد.`}
            {`

ماده ۱۰. اطلاعات شخصی کاربران شامل نام، شماره‌ی تلفن و رمز عبور آنها نزد دنج محفوظ است و آن را به شخص یا شرکت دیگری واگذار نخواهد کرد.`}
            {`

ماده ۱۱. دنج بر مبنای قوانین جرايم رایانه‌ای متعهد می‌شود با استفاده از تمامی امکانات و منابع خود در جهت حفظ و حراست از اطلاعات شخصی کاربران اقدام نماید. بدیهی است که استفاده‌ی مداوم از برنامه به منزله‌ی تایید شرایط سیاست‌نامه‌ی حفظ حریم خصوصی دنج است.`}
            {`

ماده ۱۲. دنج تعهدی در موثر بودن مدیتیشن برای کاربران ندارد. کیفیت اجرای تمرین، پیوسته بودن تمرین و ویژگی‌های شخصی می‌تواند نتایج را تحت تاثیر قرار دهد.`}
            {`
ماده ۱۳. در صورت داشتن هرگونه ابهام، سوالات خود را به یکی از روش‌های زیر با ما در میان بگذارید:

پشتیبانی: 02191006901

ایمیل دنج : app.denj@gmail.com

آدرس: تهران، میدان ونک، خیابان برزیل، خیابان شمس لاهیجانی، بن‌بست ۱۲، پلاک `}
          </Typography>
        </View>
      </Scroller>
    </Container>
  );
}
