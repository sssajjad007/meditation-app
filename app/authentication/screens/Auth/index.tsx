import React, { useEffect, useState } from "react";
import { View, Image, KeyboardAvoidingView, Platform } from "react-native";
import { Container, THEME, Typography } from "../../../library";
import { login } from "../../../library/authEvent";
import { LogoImage } from "../../assets";
import { AuthButton } from "../../components/AuthButton";
import { AuthInput } from "../../components/AuthInput";
import { styles } from "./styles";
import { register, login as loginUsecase } from "../../usecases";
import { AuthTitle } from "../../components/AuthTitle";
import { tStep } from "../../types";
import * as Analytics from "expo-firebase-analytics";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { SafeAreaView } from "react-native-safe-area-context";

import { useOtpVerify } from "./useOtpVerify";

export function Authentication() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const [phone, setPhone] = useState<string>("");
  const [genCode, setGenCode] = useState<string>("");
  const [step, setStep] = useState<tStep>("phone");
  const [error, setError] = useState<string>("");
  const [otpError, setOtpError] = useState<string>("");
  const [show, setShow] = useState<Boolean>(true);
  const [name, setName] = useState<string>("");
  async function nameEvent() {
    try {
      await Analytics.logEvent("registeration_page2_view");
    } catch (error) {
      // console.log(error);
    }
  }
  async function phoneNumberEvent() {
    try {
      await Analytics.logEvent("registeration_page1_view");
    } catch (error) {
      // console.log(error);
    }
  }
  async function onRegisterPress() {
    const registered = await register(phone);
    if (registered) {
      setStep("genCode");
    }
  }
  async function onLoginPress() {
    const loggedin: string = await loginUsecase(phone, genCode);
    if (loggedin === "error") {
      setOtpError("کد تایید اشتباهه!");
      return;
    }
    if (loggedin === "COMPLETE_USER") {
      login();
      return;
    }
    if (loggedin === "NEW_USER") {
      setStep("name");
      await nameEvent();
      return;
    }
  }
  useEffect(() => {
    phoneNumberEvent();
  }, []);
  useEffect(() => {
    if (genCode.length === 4) {
      onLoginPress();
    }
  }, [genCode]);
  useOtpVerify(setGenCode);
  return (
    <SafeAreaView style={{ flex: 1, backgroundColor: THEME.COLORS.BACKGROUND }}>
      <View style={styles.logoContainer}>
        <Image
          resizeMode={"stretch"}
          style={styles.imageSize}
          source={LogoImage}
        />
      </View>
      <View style={styles.textContainer}>
        <AuthTitle step={step} />
      </View>
      <View style={styles.inputContainer}>
        <AuthInput
          error={error}
          otpError={otpError}
          setError={setError}
          setStep={setStep}
          step={step}
          phone={phone}
          genCode={genCode}
          setGenCode={setGenCode}
          setPhone={setPhone}
          setShow={setShow}
          name={name}
          setName={setName}
        />
      </View>
      <KeyboardAvoidingView
        style={styles.buttonContainer}
        behavior={Platform.OS === "android" ? "height" : "padding"}
      >
        <AuthButton
          error={error}
          phone={phone}
          setStep={setStep}
          onLoginPress={onLoginPress}
          onRegisterPress={onRegisterPress}
          step={step}
          name={name}
          setName={setName}
        />
        {step === "phone" ? (
          <Typography
            mode={"regularSub12"}
            style={{ paddingTop: 10, color: THEME.COLORS.LABEL.SECONDARY }}
          >
            با ثبت نام و ورود با
            <Typography
              onPress={() => {
                navigation.push("TermsScreen");
              }}
              mode={"boldSub12"}
              style={{ color: THEME.COLORS.PRIMARY.NORMAL }}
            >
              {" "}
              قوانین دنج{" "}
            </Typography>
            موافقت می کنید.
          </Typography>
        ) : undefined}
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}
