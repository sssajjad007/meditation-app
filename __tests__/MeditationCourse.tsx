import React from "react";
import renderer from "react-test-renderer";
import { HomeScreen } from "../app/homePage/screens/HomeScreen";
import { MeditationCourse } from "../app/meditation/screens/MeditationCourse";

test("renders correctly", () => {
  const tree = renderer.create(<MeditationCourse />).toJSON();
  expect(tree).toMatchSnapshot();
});
