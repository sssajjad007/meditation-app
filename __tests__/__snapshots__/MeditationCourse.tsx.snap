// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    headerTitle=""
    style={
      Array [
        Object {
          "backgroundColor": "#F9FAFC",
          "flex": 1,
          "paddingBottom": 40,
          "paddingTop": 32,
        },
        undefined,
        undefined,
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Array [
          undefined,
          undefined,
        ]
      }
      style={
        Array [
          Object {
            "backgroundColor": "#F9FAFC",
            "flex": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "minHeight": 700,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "borderRadius": 8,
                "height": 270,
                "overflow": "hidden",
                "width": 675,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "height": 40,
                "marginTop": 16,
                "width": 675,
              }
            }
          >
            <Text
              mode="regularSub12"
              style={
                Array [
                  Array [
                    Object {
                      "fontFamily": "Dana-Regular",
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#676773",
                      "textAlign": "right",
                    },
                  ],
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "flex-start",
                "marginTop": 30,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "flexDirection": "row-reverse",
                  "height": 72,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 675,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flex": 4,
                    "justifyContent": "center",
                    "paddingRight": 24,
                  }
                }
              >
                <Text
                  mode="mediumBody16"
                  style={
                    Array [
                      Array [
                        Object {
                          "fontFamily": "Dana-Medium",
                          "fontSize": 16,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  
                </Text>
                <Text
                  mode="regularSub12"
                  style={
                    Array [
                      Array [
                        Object {
                          "fontFamily": "Dana-Regular",
                          "fontSize": 12,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  ۰ دقیقه
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  fadeDuration={0}
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
              </View>
              <RNGestureHandlerButton
                activeOpacity={0.105}
                collapsable={false}
                handlerTag={1}
                handlerType="NativeViewGestureHandler"
                onActiveStateChange={[Function]}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                onPress={[Function]}
                rippleColor={4293586420}
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                underlayColor="black"
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderBottomLeftRadius": undefined,
                      "borderBottomRightRadius": undefined,
                      "borderRadius": undefined,
                      "borderTopLeftRadius": undefined,
                      "borderTopRightRadius": undefined,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </RNGestureHandlerButton>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
